src/ogr/geometry.jl

```diff
@@ -1215,42 +1354,74 @@ function getgeom(geom::AbstractGeometry, i::Integer)::IGeometry
     # geometry within the container. The returned geometry remains owned by the
     # container, and should not be modified. The handle is only valid until the
     # next change to the geometry container. Use OGR_G_Clone() to make a copy.
-    if geom.ptr == C_NULL
-        return IGeometry()
-    end
+    geom.ptr == C_NULL && return IGeometry{wkbUnknown}()
     result = GDAL.ogr_g_getgeometryref(geom.ptr, i)
-    if result == C_NULL
-        return IGeometry()
-    else
-        return IGeometry(GDAL.ogr_g_clone(result))
-    end
+    result == C_NULL && return IGeometry{wkbUnknown}()
+    return IGeometry(GDAL.ogr_g_clone(result))
 end
-function getgeom(
-    geom::Union{
-        ArchGDAL.IGeometry{ArchGDAL.wkbLineString},
-        ArchGDAL.Geometry{ArchGDAL.wkbLineString},
-    }, # TODO All curves
-    i::Integer,
-)::IGeometry
-    if geom.ptr == C_NULL
-        return IGeometry()
-    end
-    return createpoint(getpoint(geom, i)[1:getcoorddim(geom)])
+# TODO create points with measures
+function getgeom(geom::AbstractGeometry{wkbLineString}, i::Integer)
+    p = getpoint(geom, i)
+    return createpoint(p[1], p[2])
+end
+function getgeom(geom::AbstractGeometry{wkbLineStringM}, i::Integer)
+    p = getpoint(geom, i)
+    return createpoint(p[1], p[2]; m=getm(geom, i))
+end
+function getgeom(geom::AbstractGeometry{wkbLineString25D}, i::Integer)
+    p = getpoint(geom, i)
+    return createpoint(p[1], p[2], p[3])
+end
+function getgeom(geom::AbstractGeometry{wkbLineStringZM}, i::Integer)
+    p = getpoint(geom, i)
+    return createpoint(p[1], p[2], p[3]; m=getm(geom, i))
```

Comment:
Can we add unit tests for these lines?

Reply:
Tested the LineString25D, and deleted the others as they were not really supported.